jax/_src/array.py

```diff
@@ -308,13 +308,6 @@ def __getitem__(self, idx):
     from jax._src.numpy import lax_numpy
     self._check_if_deleted()
 
-    if isinstance(idx, tuple):
-      num_idx = sum(e is not None and e is not Ellipsis for e in idx)
-      if num_idx > self.ndim:
-        raise IndexError(
-            f"Too many indices for array: array has ndim of {self.ndim}, but "
-            f"was indexed with {num_idx} non-None/Ellipsis indices.")
-
```

Comment:
Removed this because the logic is wrong (boolean indices `idx` consume a number of dimensions equal to `idx.ndim) and this condition is correctly checked in `_index_to_gather`.